web/server/codechecker_server/api/mass_store_run.py

```diff
@@ -191,6 +229,14 @@ def get_blame_file_data(
     return blame_info, remote_url, tracking_branch
 
 
+def checker_name_for_report(report: Report) -> Tuple[str, str]:
+    analyzer_name, checker_name = \
+        getattr(report, "analyzer_name"), getattr(report, "checker_name")
+    analyzer_name = analyzer_name if analyzer_name else "UNKNOWN"
+    checker_name = checker_name if checker_name else "NOT FOUND"
```

Comment:
Raw plists not generated to the exact format we usually consume: https://github.com/Ericsson/codechecker/blob/v6.23.0/web/tests/functional/cppcheck/test_proj/divide_zero.plist

Also handled existing behaviour:

https://github.com/Ericsson/codechecker/blob/v6.23.0/web/server/codechecker_server/api/report_server.py#L2593-L2598

https://github.com/Ericsson/codechecker/blob/v6.23.0/web/server/codechecker_server/api/mass_store_run.py#L732

Reply:
See my previous comments.

Reply:
(Side note: `previous` here is undefined, because the comments are ordered, I think, by the hunk they are attached to, irrespective of when they were typed in.)

Reply:
Also, the lack of a checker name being set to `"NOT FOUND"` was here: https://github.com/Ericsson/codechecker/blob/v6.23.0/web/server/codechecker_server/api/mass_store_run.py#L750

Now, because the string is changed to an ID that participates in a `FOREIGN KEY` lookup, and especially because it is an eagerly loaded sub-object, we have to explicitly obtain a **valid** ID, even for the unknown cases.

Unfortunately, it could also be set to **empty string** if you end up generating a `plist` which explicitly defines the corresponding key to be empty, instead of being *undefined*. We have evidence of people using CodeChecker with their own custom plist-generating infrastructure (or custom tools that somehow generate plists in the end), so saying that we only support whatever the report-converter is capable of spitting out, would also break these use cases. In the report-converter, the **checker name** is coalesced to unknown (https://github.com/Ericsson/codechecker/blob/v6.23.0/tools/report-converter/codechecker_report_converter/report/parser/plist.py#L246), but the **analyser name** is not. That was NULL-coalesced on the database level by setting `"unknown"` be the default value for the column in the schema.

```xml
<key>check_name</key><string></string>
```

Reply:
Nevertheless, I checked all the edge cases I could find and figured out a way to simplify the related logic in multiple places.